UoA-eResearch/hub-stack
# Update subhub link checker deps

## Description
Fix: This PR updates dependencies in subhub-link-checker in order to resolve security advisories, and makes changes to code in order to make it work with updated packages. It also changes Promise-based functions in subhub-link-checker into async/await functions to improve readability.

## Solution
Packages are bumped using `npm audit fix` and [Contentful's Forma 36 migration codemod](https://f36.contentful.com/migration-v3-to-v4). Manual testing and code changes were then done to address glitchy scrollbars and remove calls to deprecated methods. Unused stub component pages are removed.

More documentation is also added for deploying `subhub-link-checker`.

## Screenshots
No UI change.

## Testing
No changes requiring new unit and E2E testing.

### Manual testing
- Go to Contentful, and switch environment to dev, where these changes are applied.
- Go to the subhub He Korowai Mātauranga (or any other subhub.)
- Go to the Internal pages field. The field should be rendered with internal pages references. Select Add content > Add existing content.
- Try adding the subhub page itself. This should give a "link check failed" dialog and the item should not be added as a reference.
- Try adding another page that already belongs to another subhub, e.g. Nectar Research Cloud. This should give a "link check failed" dialog and the item should not be added as a reference.
- Try adding a page that doesn't belong to another subhub, e.g. Open Access. This should succeed.
- Publish the change and go to the corresponding topic page on `dev`. The new item should be shown.
- Remove the page reference and hit Publish to clean up.

## Have the changes been checked in the following browsers?
